src/result_generation/height/height.py

```diff
@@ -59,27 +59,38 @@ def calculate_percentile(self):
             artifact['percentile'] = cgm_error_stats_api.get_percentile_from_error_stats(
                 self.scan_meta_data_details['age'], self.scan_meta_data_details['scan_type'], self.scan_version, self.artifact_workflow_obj['name'], self.scan_workflow_obj['version'], 99)
 
-    def artifact_level_result(self, predictions, generated_timestamp, start_time):
-        """Prepare artifact level height result object"""
-        res = Bunch(dict(results=[]))
-        pos_percentile_error_99 = None
-        neg_percentile_error_99 = None
-        mae_scan = []
-        for artifact, prediction in zip(self.artifacts, predictions):
+    def calculate_scan_level_error_stats(self):
+        scan_99_percentile_pos_error = None
+        scan_99_percentile_neg_error = None
+        mae_artifact_result = []
+        for artifact in self.artifacts:
             if 'percentile' in artifact and bool(artifact['percentile']):
-                mae_scan.append(artifact['percentile']['mae'])
+                if artifact['percentile']['mae'] is not None:
+                    mae_artifact_result.append(artifact['percentile']['mae'])
                 if artifact['percentile']['99_percentile_pos_error'] is not None:
-                    if pos_percentile_error_99 is None:
-                        pos_percentile_error_99 = artifact['percentile']['99_percentile_pos_error']
+                    if scan_99_percentile_pos_error is None:
+                        scan_99_percentile_pos_error = artifact['percentile']['99_percentile_pos_error']
                     else:
-                        pos_percentile_error_99 = max(pos_percentile_error_99,
-                                                      artifact['percentile']['99_percentile_pos_error'])
+                        scan_99_percentile_pos_error = max(scan_99_percentile_pos_error,
+                                                           artifact['percentile']['99_percentile_pos_error'])
                 if artifact['percentile']['99_percentile_neg_error'] is not None:
-                    if neg_percentile_error_99 is None:
-                        neg_percentile_error_99 = artifact['percentile']['99_percentile_neg_error']
+                    if scan_99_percentile_neg_error is None:
+                        scan_99_percentile_neg_error = artifact['percentile']['99_percentile_neg_error']
                     else:
-                        neg_percentile_error_99 = min(neg_percentile_error_99,
-                                                      artifact['percentile']['99_percentile_neg_error'])
+                        scan_99_percentile_neg_error = min(scan_99_percentile_neg_error,
+                                                           artifact['percentile']['99_percentile_neg_error'])
+        if 'percentile' in artifact and bool(artifact['percentile']):
```

Comment:
here will will get the error. Look it closely. 

This statement apply in case if there is loop.
  if 'percentile' in artifact and bool(artifact['percentile']):

Reply:
Thx!!